src/main/java/io/numaproj/numaflow/sinker/Sinker.java

```diff
@@ -10,7 +10,13 @@ public abstract class Sinker {
     /**
      * method will be used for processing messages.
      * @param datum current message to be processed
-     * @return Response to indicate success or failure.
      */
-    public abstract Response processMessage(Datum datum);
+    public abstract void processMessage(Datum datum);
+
+    /**
+     * method will be used for returning the responses.
+     * each response should contain the id of the message
+     * @return ResponseList which contains the responses
+     */
+    public abstract ResponseList getResponse();
```

Comment:
Could you add descriptions on the contract between `processMessage` and `getResponse` as they are dependent on each other? Implementers need to know the contract so that they implement the method following the Actor pattern.

e.g. "N processMessage calls will generate N responses, then a getResponse call will return the N responses and clean up the list, then the new M processMessage calls will start populating the empty list until next getResponse call retrieve them again", the example might not be accurate but you get the idea.